io.github.trevorsandy.LPub3D.json

```diff
@@ -0,0 +1,274 @@
+{
+  "id": "io.github.trevorsandy.LPub3D",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15",
+  "sdk": "org.kde.Sdk",
+  "command": "lpub3d24",
+  "rename-icon": "lpub3d",
+  "rename-desktop-file": "lpub3d.desktop",
+  "rename-appdata-file": "io.github.trevorsandy.appdata.xml",
+  "build-options": {
+    "env": {
+      "LP3D_DIST_DIR_PATH": "/app/lpub3d_linux_3rdparty",
+      "LDRAWDIR": "/app/lpub3d_linux_3rdparty/ldraw"
+    }
+  },
+  "cleanup": [
+    "/app/lpub3d_linux_3rdparty"
+  ],
+  "finish-args": [
+    "--device=dri",
+    "--share=ipc",
+    "--socket=x11",
+    "--filesystem=home",
+    "--env=TMPDIR=/var/tmp",
+    "--env=QT_ENABLE_HIGHDPI_SCALING=1"
+  ],
+  "modules": [
+    "shared-modules/glu/glu-9.json",
+    {
+      "name": "freeglut",
+      "build-options": {
+        "cflags": "-fcommon"
+      },
+      "buildsystem": "cmake-ninja",
+      "cleanup": [
+        "/include",
+        "*.a"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://downloads.sourceforge.net/project/freeglut/freeglut/3.2.1/freeglut-3.2.1.tar.gz",
+          "sha256": "d4000e02102acaf259998c870e25214739d1f16f67f99cb35e4f46841399da68"
+        }
+      ]
+    },
+    {
+      "name": "mesa",
+      "buildsystem": "meson",
+      "config-opts": [
+        "-Dosmesa=classic",
+        "-Ddri-drivers=[]",
+        "-Dplatforms=['x11']",
+        "-Dgallium-drivers=[]",
+        "-Dvulkan-drivers=[]"
+      ],
+      "cleanup": [
+        "/share/drirc.d",
+        "/include",
+        "*.a"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://archive.mesa3d.org/mesa-20.2.6.tar.xz",
+          "sha256": "f12ca3c6c622f11cd79ad66b4220f04514fa96f795062fe92a37339ab19885db"
+        }
+      ]
+    }, 
+    {
+      "name": "ldraw-libraries",
+      "buildsystem": "simple",
+      "build-commands": [
+        "install -d ${FLATPAK_DEST}/lpub3d_linux_3rdparty/",
+        "mv -f complete.zip ${FLATPAK_DEST}/lpub3d_linux_3rdparty/",        
+        "mv -f lpub3dldrawunf.zip ${FLATPAK_DEST}/lpub3d_linux_3rdparty/",
+        "mv -f tenteparts.zip ${FLATPAK_DEST}/lpub3d_linux_3rdparty/",
+        "mv -f vexiqparts.zip ${FLATPAK_DEST}/lpub3d_linux_3rdparty/",
+        "unzip -d ${FLATPAK_DEST}/lpub3d_linux_3rdparty -q ${FLATPAK_DEST}/lpub3d_linux_3rdparty/complete.zip"
```

Comment:
I do have question about this section:
1. the libraries are copied to the user config directory. Which mean that now `~/.var/app/io.github.trevorsandy.LPub3D/data` is 506MB. It contains `complete.zip` and `lpub3dldrawunf.zip`, and the unzip `ldraw`
2. we unzip ldraw here and it end up in the flatpak. And it seems to not be in the search path for the library.

So isn't there a way that would avoid not having this duplicated.

Reply:
> I do have question about this section:

The LDraw folder should be removed at cleanup.

There is...

```
"cleanup": [
     "/app/lpub3d_linux_3rdparty"
   ],
```

... is in the manifest but I see that the staging folder (containing LDraw, and a lot of other unneeded content. The bulk of which make up the 500 MB data size) is not removed.

Cheers,

Reply:
Cleanup should be 
```
"cleanup": [
     "/lpub3d_linux_3rdparty"
   ],
```
It's all based on the prefix (which is `/app`)

Reply:
> Cleanup should be

Very well. I’ll update the manifest when I add the appdata file. 

> It's missing an appstream file.

I noticed some manifests do not include any appdata file placement reference while others use either `/app/share/appdata` or `/app/share/metainfo`. I saw the documentation say`/app/share/metainfo` should be used however. What is the difference and moreover, how come some manifesta do not have a placement reference and the build is not affected ?

Cheers,

Reply:
`/app/share/appdata` is deprecated. It still work, but whenever the new one should be used.
And some package install it by default so the manifest doesn't need to handle it.

Reply:
`${FLATPAK_DEST}/opt` is fine, but you could as well install everything into `${FLATPAK_DEST}` since it all yours anyway.